sbi/simulators/mp_simulator.py

```diff
@@ -0,0 +1,279 @@
+import multiprocessing as mp
+import numpy as np
+from tqdm.auto import tqdm
+from typing import Callable
+import torch
+from sys import platform
+
+
+def simulate_mp(
+    simulator: Callable,
+    theta: torch.Tensor,
+    num_workers: int = 4,
+    worker_batch_size: int = 20,
+    verbose: bool = False,
+    show_progressbar: bool = True,
+) -> (torch.Tensor, torch.Tensor):
+    """
+    Wrapper function for the callable simulator to allow it to use multiprocessing.
```

Comment:
"Return parameters theta and simulated data x, simulated using multiprocessing on the passed simulator and parameters theta."